mlflow/azureml/__init__.py

```diff
@@ -380,16 +382,34 @@ def deploy(
             _copy_file_or_tree(src=absolute_model_path, dst=model_directory_path),
         )
 
-        registered_model = AzureModel.register(
-            workspace=workspace, model_path=tmp_model_path, model_name=model_name, tags=tags
-        )
-
-        _logger.info(
-            "Registered an Azure Model with name: `%s` and version: `%s`",
-            registered_model.name,
-            registered_model.version,
-        )
-
+        registered_model = None
+
+        try:
+            if model_uri.startswith("models:/"):
+                m_name = model_uri.split('/')[-2]
+                m_version = int(model_uri.split('/')[-1])
+                m_id = "{}:{}".format(m_name, m_version)
+                registered_model = AzureModel(workspace, id=m_id)
+
+                _logger.info("Found registered model in AzureML with ID '{}'".format(m_id))
+            elif model_uri.startswith("runs:/") and get_tracking_uri().startswith("azureml:/"):
```

Comment:
We should check for startswith "azureml:" although this may not be compatible with our target of "azureml", 

startswith("azureml:") or  =="azureml" would be longer lived